zulipterminal/ui_tools/boxes.py

```diff
@@ -1124,22 +1126,32 @@ def reactions_view(self, reactions: List[Dict[str, Any]]) -> Any:
                     user_name = "You"
                 reaction_stats[reaction["emoji_name"]].append((user_id, user_name))
 
-            for reaction, ids in reaction_stats.items():
-                if (my_user_id, "You") in ids:
-                    ids.remove((my_user_id, "You"))
-                    ids.append((my_user_id, "You"))
-
-            reaction_texts = [
-                (
-                    "reaction_mine"
-                    if my_user_id in [id[0] for id in ids]
-                    else "reaction",
-                    f" :{reaction}: {len(ids)} "
-                    if len(reactions) > TOTAL_USERNAMES_VISIBLE
-                    else f" :{reaction}: {', '.join([id[1] for id in ids])} ",
-                )
-                for reaction, ids in reaction_stats.items()
-            ]
+            if len(reactions) <= TOTAL_USERNAMES_VISIBLE:
+                for reaction, ids in reaction_stats.items():
+                    if (my_user_id, "You") in ids:
+                        ids.remove((my_user_id, "You"))
+                        ids.append((my_user_id, "You"))
+                        reaction_texts.append(
+                            (
+                                "reaction_mine",
+                                f" :{reaction}: {', '.join([id[1] for id in ids])} ",
+                            )
+                        )
+                    else:
+                        reaction_texts.append(
+                            (
+                                "reaction",
+                                f" :{reaction}: {', '.join([id[1] for id in ids])} ",
+                            )
```

Comment:
Note that you can simplify the above by bringing the `append` statements out of the conditionals using `reaction_style` for `"reaction"` and`"reaction-mine"`.

This can be added when the feature itself was introduced in the previous commit but I'd wait for @neiljp to confirm if this was what he imagined.